source/platforms/bitbucket_cloud/_tests_/_bitbucket_cloud_git.test.ts

```diff
@@ -0,0 +1,226 @@
+      "lib/components/gene/biography.js",
+      "lib/components/related_artists/index.js",
+      "lib/components/related_artists/related_artist.js",
+    ])
+
+    expect(gitJSONDSL.deleted_files).toEqual([
+      "lib/components/artist/related_artists/index.js",
+      "lib/components/artist/related_artists/related_artist.js",
+      "lib/components/gene/about_gene.js",
+    ])
+  })
+
+  it("shows the diff for a specific file", async () => {
+    const { diff } = (await gitDSL.diffForFile("tsconfig.json"))!
+
+    expect(stripWhitespaceForSnapshot(diff)).toMatchSnapshot()
+  })
+
+  it("should include `before` text content of the file", async () => {
+    const { before } = (await gitDSL.diffForFile("tsconfig.json"))!
+
+    expect(stripWhitespaceForSnapshot(before)).toMatchSnapshot()
+  })
+
+  it("should include `after` text content of the file", async () => {
+    const { after } = (await gitDSL.diffForFile("tsconfig.json"))!
+
+    expect(stripWhitespaceForSnapshot(after)).toMatchSnapshot()
+  })
+
+  it("should include `added` text content of the file", async () => {
+    const { added } = (await gitDSL.diffForFile("tsconfig.json"))!
+
+    expect(stripWhitespaceForSnapshot(added)).toMatchSnapshot()
+  })
+
+  it("should include `removed` text content of the file", async () => {
+    const { removed } = (await gitDSL.diffForFile("tsconfig.json"))!
+
+    expect(stripWhitespaceForSnapshot(removed)).toMatchSnapshot()
+  })
+
+  it("resolves to `null` for files not in modified_files", async () => {
+    const result = (await gitDSL.diffForFile("fuhqmahgads.json"))!
+
+    expect(result).toBeNull()
+  })
+
+  it("shows the structured diff for a specific file", async () => {
+    const { chunks } = (await gitDSL.structuredDiffForFile("tsconfig.json"))!
+    expect(chunks).toMatchSnapshot()
+  })
+
+  it("sets up commit data correctly", async () => {
+    const exampleCommit: GitCommit = {
+      author: {
+        date: "2019-05-13T13:24:53+00:00",
+        email: "foo@bar.com",
+        name: "Hello Core",
+      },
+      committer: {
+        date: "2019-05-13T13:24:53+00:00",
+        email: "foo@bar.com",
+        name: "Hello Core",
+      },
+      message: "Merged in develop (pull request #174)\n\nDevelop to SIT\n\nApproved-by: Hello Core <foo@bar.com>\n",
+      parents: ["77fdf99fcc08dce8bd49c9aedfc7442f2bd40bde", "a584f1256b8522cfce4ccc33dfbf157590a6044d"],
+      sha: "7f73207cea935b2fd07a76028484c20eb18422ba",
+      tree: null,
+      url: "https://bitbucket.org/foo/bar/commits/7f73207cea935b2fd07a76028484c20eb18422ba",
+    }
+
+    expect(gitDSL.commits[0]).toEqual(exampleCommit)
+  })
+
+  // it("writes a JSON DSL fixture", async () => {
+  //   const fakeSource = new FakeCI({})
+  //   const dataSent = await jsonDSLGenerator(bbc, fakeSource, {} as any)
+  //   dataSent.settings.github.accessToken = "12345"
+
+  //   writeFileSync(pathJoin(fixtures, "bbc-dsl-input.json"), JSON.stringify(dataSent, null, "  "), "utf8")
```

Comment:
This caused `jest --watch` loop infinitely, and I don't know how to fix it. 😢

Reply:
No worries, you can add it to the [watch ignore](https://jestjs.io/docs/en/configuration#watchpathignorepatterns-array-string) in Jest

Reply:
Thank you! 
I found that it happened because `test-results.json` keeps changing, so I added it to `watchPathIgnorePatterns`.